torchmdnet/datasets/spice.py

```diff
@@ -1,21 +1,50 @@
-import imp
+import hashlib
 import h5py
+import json
 import numpy as np
 import os
 import torch as pt
-from torch_geometric.data import Data, Dataset
+from torch_geometric.data import Data, Dataset, download_url
 from tqdm import tqdm
 
 
 class SPICE(Dataset):
 
     """
     SPICE dataset (https://github.com/openmm/spice-dataset)
+
+    The dataset consist of several subsets (https://github.com/openmm/spice-dataset/blob/main/downloader/config.yaml). By default all the subsets are loaded or they can be selected with `dataset_arg`. For example, this loads just two subsets:
+
+    >>> ds = SPICE('.', dataset_arg='{"subsets": ["SPICE PubChem Set 1 Single Points Dataset v1.2", 'SPICE PubChem Set 2 Single Points Dataset v1.2']}')
```

Comment:
Why pass a string of a json instead of a python dict?

Reply:
Because the argument parser doesn't support nested dictionaries: https://github.com/torchmd/torchmd-net/blob/080cf2214b146f8c92230d34dc3435136e47a604/scripts/train.py#L20-L92

Reply:
I would still make the class accept python dicts. Just `json.parse` the string of argparse if you are using command line to convert it to python dict and pass that to the class. Makes the python API layer more usable

Reply:
Other loaders expect `loader_arg` to be `str`. How should I handle that?

Reply:
I looked at the other loaders. They accept single strings of undocumented parameters so I don't feel like it's a standard. Like here where it's silently converted to `paths` https://github.com/torchmd/torchmd-net/blob/main/torchmdnet/datasets/qm9q.py#L35-L61. I would argue that the CLI should always accept a json string even for the old datasets (like `'{"paths": "/tmp/"}'` and convert it to python dicts and the old datasets should be modified to read the specific key/value from the dict. That makes it also clearer what arguments are being set from command line.

Reply:
@stefdoerr Yes, it would be a good solution. Could you make a PR with the proposed changes for the old loaders? After I'll follow your lead with this PR.